File: AndroidStudioProjects/CourtCounter/app/src/main/java/com/example/android/courtcounter/MainActivity.java
package com.example.android.courtcounter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    int Ascore=0;int Bscore=0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void pointA3(View view) {
        Ascore=Ascore+3;
        displayForTeamA(Ascore);
    }
    public void pointA2(View view) {
        Ascore=Ascore+2;
        displayForTeamA(Ascore);
    }
    public void freeThrowA(View view) {
        Ascore=Ascore+1;
        displayForTeamA(Ascore);
    }

    private void displayForTeamA(int ascore) {
        TextView scoreTeamA=findViewById(R.id.score_team_A);
        scoreTeamA.setText(String.valueOf(ascore));
    }
    private void displayForTeamB(int bscore) {
        TextView scoreTeamB=findViewById(R.id.score_team_B);
        scoreTeamB.setText(String.valueOf(bscore));
    }
    public void pointB3(View view) {
        Bscore=Bscore+3;
        displayForTeamB(Bscore);
    }
    public void pointB2(View view) {
        Bscore=Bscore+2;
        displayForTeamB(Bscore);
    }public void freeThrowB(View view) {
        Bscore=Bscore+1;
        displayForTeamB(Bscore);
    }
     public void resetScore(View view) {
        Ascore=0;
        Bscore=0;
        displayForTeamA(Ascore);
        displayForTeamB(Bscore);
    }
}

File: RetrofitDemo/app/src/main/java/com/example/retrofitdemo/MainActivity.kt
package com.example.retrofitdemo

import android.app.ProgressDialog.show
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import android.widget.Toast
import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import androidx.lifecycle.liveData
import retrofit2.Response

class MainActivity : AppCompatActivity() {
    private lateinit var retService: AlbumService
    private lateinit var textView: TextView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        retService = RetrofitInstance.getRetrofitInstance().create(AlbumService::class.java)
        textView = findViewById<TextView>(R.id.text_view)
        //getRequestWithQueryParameters()
        //getRequestWithPathParameters()
        uploadAlbum()
    }


    private fun getRequestWithQueryParameters() {
        val responseLiveData: LiveData<Response<Albums>> = liveData {
            val response = retService.getSortedAlbums(3)
            emit(response)
        }
        responseLiveData.observe(this, Observer {
            val albumsList = it.body()?.listIterator()

            if (albumsList != null) {
                while (albumsList.hasNext()) {
                    val albumsItem = albumsList.next()
                    val result = " " + "Album Title : ${albumsItem.title}" + "\n" +
                            " " + "Album id : ${albumsItem.id}" + "\n" +
                            " " + "User id : ${albumsItem.userId}" + "\n\n\n"
                    textView.append(result)
                }
            }

        })
    }

    private fun uploadAlbum(){
        val album = AlbumsItem(101,"My title" , 3)
        val postResponse : LiveData<Response<AlbumsItem>> = liveData {
            val response = retService.uploadAlbum(album)
            emit(response)
        }

        postResponse.observe(this, Observer {
            val receivedAlbumsItem = it.body()
            val result = " " + "Album Title : ${receivedAlbumsItem?.title}" + "\n" +
                    " " + "Album id : ${receivedAlbumsItem?.id}" + "\n" +
                    " " + "User id : ${receivedAlbumsItem?.userId}" + "\n\n\n"
            textView.text = result
        })

    }

    private fun getRequestWithPathParameters(){
        //path parameter
        val pathResponse: LiveData<Response<AlbumsItem>> = liveData {
            val response = retService.getAlbum(3)
            emit(response)
        }

        pathResponse.observe(this, Observer {
            val title = it.body()?.title
            Toast.makeText(applicationContext, title, Toast.LENGTH_LONG).show()
        })
    }

}